go/vt/vtgateproxy/vtgateproxy.go

```diff
@@ -59,32 +60,36 @@ func (proxy *VTGateProxy) getConnection(ctx context.Context, target string) (*vt
 	// If the connection exists, return it
 	proxy.mu.RLock()
 	existingConn := proxy.targetConns[target]
+	proxy.mu.RUnlock()
+
 	if existingConn != nil {
-		proxy.mu.RUnlock()
 		log.V(100).Infof("Reused connection for %v\n", target)
 		return existingConn, nil
 	}
 
 	// No luck, need to create a new one. Serialize new additions so we don't create multiple
 	// for a given target.
 	log.V(100).Infof("Need to create connection for %v\n", target)
-	proxy.mu.RUnlock()
+
 	proxy.mu.Lock()
+	defer proxy.mu.Unlock()
 
-	// Otherwise create a new connection after dropping the lock, allowing multiple requests to
-	// race to create the conn for now.
-	grpcclient.RegisterGRPCDialOptions(func(opts []grpc.DialOption) ([]grpc.DialOption, error) {
-		return append(opts, grpc.WithDefaultServiceConfig(`{"loadBalancingConfig": [{"round_robin":{}}]}`)), nil
-	})
+	// Check again in case conn was made between lock acquisitions.
+	existingConn = proxy.targetConns[target]
+	if existingConn != nil {
+		log.V(100).Infof("Reused connection for %v\n", target)
+		return existingConn, nil
+	}
 
+	// Otherwise create a new connection. TODO: confirm this doesn't actually make a TCP connection, and returns quickly,
```

Comment:
Ya - confirmed, the model in grpc is pretty weird, you create a "conn", but that's basically just a handle to make RPC requests. It's not until you make an RPC call that it will actually start mapping the RPC to a subconn (which maps to a TCP connection).

That's also why you can reuse the conn - it should represent a logical target to send the RPC commands to.